gitlab-pipeline/stage/test.yml

```diff
@@ -731,7 +743,41 @@ test_full_integration_enterprise:
     reports:
       junit: results_full_integration.xml
 
-test_full_integration_open_source:
-  extends: test_full_integration_enterprise
+
+test:integration:open_source:build_client:
+  extends: .integration_setup_template
+  only:
+    variables:
+      - ( $RUN_INTEGRATION_TESTS == "true" && $BUILD_CLIENT == "true" )
+  needs:
+    - init_workspace
+    - build_servers
+    - build_client
+    - build_client_docker
+    - build_mender-artifact
+
+
+test:integration:enterprise:build_client:
+  extends: test:integration:open_source:build_client
   variables:
-    TEST_SUITE: "open"
+    TEST_SUITE: "enterprise"
+
+
+test:integration:open_source:without_build_client:
+  extends: .integration_setup_template
+  only:
+    variables:
+      - $RUN_INTEGRATION_TESTS == "true"
+  except:
+    variables:
+      - $BUILD_CLIENT == "true"
+  needs:
+    - init_workspace
+    - build_servers
+    - build_mender-artifact
+
+
+test:integration:enterprise:without_build_client:
```

Comment:
Mmmm, at this point seems to me that build_client/without_build_client should take precedence over open_source/enterprise. Thoughts?

Reply:
Sorry, not following. Do you mean that the name should be different?

Reply:
Just thinking out loud, but I think `test:integration:without_build_client:enterprise:` is better than `test:integration:enterprise:without_build_client:`.

When reading the file, this job comes the last one and modifies the os/enterprise flavor, hence I think it is best.

This is really minor, both approaches are good!

Reply:
Once again, I concur!
Changing! Thx

Reply:
On a roll! I'm going to be bold, and say that I'm not sure `build_client` is the best name, and shoot:
* `fresh_client`
* `source_client`
* `new_client`

instead. Agree with any of them?

Reply:
@oleorhagen `source_client` gets my vote